jvm/src/main/java/com/microsoft/bond/protocol/UnsignedHelper.java

```diff
@@ -2,29 +2,134 @@
 
 import java.math.BigInteger;
 
-public class UnsignedHelper {
-    private final static BigInteger TWO_TO_64 = BigInteger.valueOf(2L).pow(64);
-    private final static BigInteger LONG_MAX_PLUS_ONE = BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE);
+/**
+ * Contains helper methods for working with unsigned integers represented by signed types.
+ * Since Java doesn't have unsigned integral types, they are represented by signed integral
+ * types of the same width (e.g. short for unsigned 16-bit integer) where the sign bit is
+ * interpreted as the most significant bit of the unsigned value. Please note that applications
+ * using unsigned integers in numeric expressions are responsible for implementing overflow
+ * protection such as type promotion (e.g. use int to represent unsigned 16-bit value).
+ */
+final class UnsignedHelper {
```

Comment:
We may make it public when/if needed. What kind of javadoc woudl you like to see that is not already there? I am not sure I understand what you mean by "type migration". Other classes in the package use this utility (e.g. when doing bit operations on uint16 and need to explicitly convert to int32 to avoid implicit casts).

---
In reply to: [116110091](https://github.com/tstein/bond/pull/4#discussion_r116110091) [](ancestors = 116110091)